qlib/utils/data.py

```
@@ -115,3 +116,14 @@ def guess_horizon(label: List):
     expr = DatasetProvider.parse_fields(label)[0]
     lft_etd, rght_etd = expr.get_extended_window_size()
     return rght_etd
+
+
+def update_config_with_validation(base_config: dict, ext_config: Union[dict, List[dict]], 
+                                  validate: bool = True) -> dict:
+    updated_config = update_config(base_config, ext_config)
+    
+    if validate:
+        if not validate_config(updated_config):
```

Ruff (0.11.9)

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)